### PR TITLE
Add graph data class and two examples

### PR DESCRIPTION
**Pull request by Hzaatiti**

### CODE REVIEW

File: geomstats/learning/graph_data_preparation.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        walk_length : int
+            length of a random walk in terms of number of edges
+
+        number_walks_per_node : int
+            number of generated walks starting from each node of the graph
+
+        Returns
+        -------
+        self : array-like,
+            shape=[number_walks_per_node*len(self.edges), walk_length]
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/learning/graph_data_preparation.py]: Maybe add:

```self.n_edges```

as an attribute?

File: geomstats/learning/graph_data_preparation.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        paths = gs.empty((0, walk_length + 1), dtype=int)
+        for index in range(0, len(self.edges)):
+            for i in range(number_walks_per_node):
+                paths = gs.vstack((paths, self._walk(index)))
+        self.paths = paths
+        return paths
+
+    def _walk(self, index):
+        """Generate a single random walk."""
+        path = gs.array([], dtype=int)
```
Comment (ninamiolane) [inline geomstats/learning/graph_data_preparation.py]: I think you could use a list directly therE?

File: tests/test_load_graph.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from geomstats.learning.graph_data_preparation import Graph
+
+
+class TestLoadDefaultGraph(geomstats.tests.TestCase):
+    """Test for loading graph-structured data."""
+
+    @geomstats.tests.np_only
+    def setUp(self):
+        """Declare the graph by default and the Karate club graph."""
+        self.G1 = Graph()
```
Comment (ninamiolane) [inline tests/test_load_graph.py]: no uppercase in variables names

File: tests/test_load_graph.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        self.G2 = Graph(
+            Graph_Matrix_Path='examples\\data_example\\'
+                              'graph_karate\\Karate.txt',
+            Labels_Path='examples\\data_example\\'
+                        'graph_karate\\Karate_Labels.txt')
+
+    @geomstats.tests.np_only
+    def test_graph_load(self):
```
Comment (ninamiolane) [inline tests/test_load_graph.py]: Nice!

File: geomstats/learning/graph_data_preparation.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+"""Prepare and Process Graph-structured data."""
+
+import random
+
+import geomstats.backend as gs
+
+
+class Graph():
```
Comment (ninamiolane) [inline geomstats/learning/graph_data_preparation.py]: `class Graph:` works in python3 I think

File: geomstats/learning/graph_data_preparation.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    Graph_Matrix_Path : string
+        Path to graph adjacency matrix.
+
+    Labels_Path : string
+        Path to labels of the nodes of the graph.
+    """
+
+    edges = None
+    labels = None
+    paths = None
```
Comment (ninamiolane) [inline geomstats/learning/graph_data_preparation.py]: Why are paths, walk_length and n_walks_per_node class attributes?

**Comment by Hzaatiti** [comment]
Is there a reason why the coverage is dropping by 38% ? is it because the test i made is failing ? (file not found error)

**Comment by ninamiolane** [comment]
Yes:
FileNotFoundError: [Errno 2] No such file or directory: 'examples\\data\\graph_random\\graph_random.txt'

- Either the double \\ is a problem,
- or the file is not uploaded for travis.

To test the second option, maybe create another PR with only the data files. I'll merge it, and then this second PR  loading the files.

The coverage drops because the data files are large and there is no coding running through them... which is normal. I will look for a way to exclude data files from code coverage.

Another remark: could you put the graph_data_preparation.py in a new folder, that we call `datasets`, such that we have:

- geomstats/geometry
- geomstats/learning
- geomstats/datasets : for all functions that prepare datasets?

Thank you!

**Comment by ninamiolane** [comment]
ImportError: No module named 'geomstats.learning.graph_data_preparation' ;)

File: geomstats/datasets/graph_data_preparation.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
+"""Prepare and process graph-structured data."""
+
+import random
+
+import geomstats.backend as gs
+
+DEFAULT_GRAPH_MATRIX_PATH = r'examples\data' \
```
Comment (ninamiolane) [inline geomstats/datasets/graph_data_preparation.py]: Backslashs / and not \ ?

**Pull request merged**